### PR TITLE
Feature/jdbc

### PR DESCRIPTION
**Pull request by PeterLappo**
Hi,
I've added some system information to the prestodb JDBC driver so that you can see schema, table and column data in tools like dbeaver which are Java tools that use the presto JDBC driver. I'm presuming but haven't tested it that it will also work with presto ODBC drivers.
Here are a couple of dbeaver screen shots.
dbeaver1
dbeaver2

Thanks

### CODE REVIEW

File: dask_sql/server/app.py
Comment (charlesbluca) [inline dask_sql/server/app.py:79]: If this is mandatory for compatibility with JDBC driver and is always occurring, perhaps we should just perform the adjustment directly here?

File: tests/integration/test_jdbc.py
Comment (charlesbluca) [inline tests/integration/test_jdbc.py:72]: Wondering if there is a way to place these assertions in a helper function like `assert_resp` to simplify the tests; okay if this isn't possible

File: tests/integration/test_server.py
Comment (charlesbluca) [inline tests/integration/test_server.py:16]: Any reason why we're manually creating a context for the standard server tests?

**Review by charlesbluca — COMMENTED** [review]
Thanks for the work here @PeterLappo 😄 a couple comments:

Also looks like this PR is failing style checks, this should be resolved by running `pre-commit run --all-files` on your branch and pushing the resulting changes

File: dask_sql/server/presto_jdbc.py
Comment (PeterLappo) [reply dask_sql/server/presto_jdbc.py:124]: This is what the presto driver expects. I thought it odd too

File: dask_sql/server/presto_jdbc.py
Comment (PeterLappo) [reply dask_sql/server/presto_jdbc.py:28]: Fair, will update...

File: dask_sql/server/app.py
Comment (PeterLappo) [reply dask_sql/server/app.py:185]: I did consider a flag, but makes sense, and will disable by default

File: tests/integration/test_jdbc.py
Comment (PeterLappo) [reply tests/integration/test_jdbc.py:155]: I come from a Java world where the convention is to put "private" code at the end but actually coped from test_server.py which has it at the end. Shall I re-use from test_server.py? @charlesbluca

File: tests/integration/test_server.py
Comment (PeterLappo) [reply tests/integration/test_server.py:16]: The test fails if you run `pytest tests/integration/test_server.py` in isolation. And as I used this module as starter for test_jdbc.py I discovered after some messing about that tests/integration/test_compatibility.py  has to run first to ensure test_server.py and test_jdbc.py could run. It means both will now run independently.

**Comment by PeterLappo** [comment]
@charlesbluca Updated as recommended. What is the next step to merge to main?

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

**Comment by charlesbluca** [comment]
add to allowlist

File: dask_sql/server/presto_jdbc.py
Comment (charlesbluca) [reply dask_sql/server/presto_jdbc.py:99]: Ah okay thanks for the clarification - if Trino is the more active fork of Presto, perhaps it would be worthwhile to explore if it is possible to use here instead of PrestoDB, though there is no push to do that before merging this in

File: tests/integration/test_jdbc.py
Comment (charlesbluca) [reply tests/integration/test_jdbc.py]: Ah I didn't realize that this was also the case in test_server.py; in that case my preference would be to move the util functions up to the top in both files - my reasoning here being to match up with other testing files (such as [test_compatibility.py](https://github.com/dask-contrib/dask-sql/blob/main/tests/integration/test_compatibility.py)) that place their utility functions at the top of the file.

I also note that since we are duplicating functions like `get_result_or_error`, we might want to add a testing utility module as a central location for functions like this, but that is out of the scope of this PR.

File: tests/integration/test_server.py
Comment (charlesbluca) [reply tests/integration/test_server.py:16]: That makes sense, thanks for the explanation

File: dask_sql/server/presto_jdbc.py
Comment (charlesbluca) [inline dask_sql/server/presto_jdbc.py:36]: If we leave the commented code, maybe we should format this as a `TODO` to indicate that future work can be done to enable catalogs:

```suggestion
    # TODO: add support for catalogs in presto interface
    # see https://github.com/dask-contrib/dask-sql/pull/351
```

File: dask_sql/server/app.py
Comment (charlesbluca) [inline dask_sql/server/app.py:86]: Just trying to cut down this explanation; feel free to let me know if I've missed something here

```suggestion
        # required for PrestoDB JDBC driver compatibility
        # replaces queries to unsupported `system` catalog with queries to `system_jdbc`
        # schema created by `create_meta_data(context)` when `jdbc_metadata=True`
        # TODO: explore Trino which should make JDBC compatibility easier but requires
        # changing response headers (see https://github.com/dask-contrib/dask-sql/pull/351)
```

**Review by charlesbluca — COMMENTED** [review]
Thanks for addressing my concerns @PeterLappo 🙂 would you mind merging the latest changes? Believe the last test failures were due to some changes in upstream libraries which should be resolved now

**Comment by PeterLappo** [comment]
@charlesbluca have merged with main. Does CI pipeline run automatically?

**Comment by charlesbluca** [comment]
Thanks!

> Does CI pipeline run automatically?

It normally does, but since you haven't pushed any commits to this repo your PR requires approval for CI to run - I've approved and this should be good to merge if tests pass

**Review by charlesbluca — APPROVED** [review]

**Pull request merged**
